.github/workflows/ci.yml

```diff
@@ -736,8 +736,65 @@ jobs:
   # Jobs that build and run IREE e2e tests/benchmarks                          #
   ##############################################################################
 
+  build_tracy_tools:
```

Comment:
I'd rather we not add another job to ci.yml just for building Tracy tools:
* This is 100+ lines of yml to build an infrequently changing tool
* This adds yet another node to the already complex graph in ci.yml's Summary view (moving to a separate workflow file would help)
* This adds yet another check to all PRs (44 successful, 2 skipped on this PR)
* This takes ~2 minutes to run on a large machine (https://github.com/iree-org/iree/actions/runs/4077267899/jobs/7026083060)

Can we instead either:
* include this in a Docker image
* store it in GCS
* build it as part of another CI step

---

More generally, I don't like how we're supporting a build configuration for a third-party tool (`IREE_BUILD_TRACY`). Tracy already provides binary releases for some platforms (Windows), and we might be able to contribute builds for other platforms that other users could also benefit from.

Reply:
The main reason to build the capture tool in CI is so that we don't need to rebuild it manually when bumping the version. Especially building the arm64 version to run on RPi. But given we rarely bump the tracy, a proper way is probably to have another workflow that can be manually triggered to compile and upload tracy binaries.

For now I'm fine to upload the prebuilt binaries in GCS (as we are doing for RPi now).

It could take some work to release linux binaries in the upstream, given the diversity of linux distribution and versions of libraries (libcapstone, libtbb). Statically linking the libraries can solve many compatibility issues, but we also need to release cross-compiled binaries, for example, IREE runs tracy-capture on RPi. But this is a good improvement in the future.

Reply:
I'd rather take the complexity hit when we do update (about once every two months):
```
λ git log --pretty=format:"%h %ad %s" --date=short -- third_party/tracy
479c1bba7 2023-01-25 Bump tracy to 52b6af88 (2023-01-25) (#11954)
a76e13003 2023-01-19 Update to wolfpld/tracy@baaed68b (2023-01-19) (#11909)
f05f51665 2022-08-03 Update Tracy to eeffb6d2 (2022-08-02) (#9991)
29c6ab50d 2022-08-02 Revert "Update Tracy to 5a36782c (2022-08-01) (#9977)" (#9986)
25bead318 2022-08-01 Update Tracy to 5a36782c (2022-08-01) (#9977)
af853b11c 2022-04-22 bump tracy to 0.8.1 (#8957)
68fbba015 2022-04-14 Update Tracy to 6998546f to pull in @bjacob's fixes (#8898)
450f0c80a 2022-03-24 bump tracy (#8629)
502db411d 2021-09-16 Update Tracy to latest @058e8901 (#7070)
790ed1ad9 2021-07-17 Bumping tracy submodule to 0.7.9+. (#6475)
893f1a2d1 2021-05-10 Bumping tracy and defining TRACY_NO_FRAME_IMAGE. (#5815)
494fe32d7 2021-03-31 Bumping the tracy version to 0.7.7 (WIP). (#5272)
33c4e3946 2021-03-12 Bump Tracy submodule to v0.7.6. (#5094)
c4129446f 2020-12-01 Bump Tracy to get Ben's dynamic zone colors and to get past (#4042)
fc8da0352 2020-11-26 Update Tracy to get clean shutdown on Android (#4015)
89ef63284 2020-11-23 bump Tracy to get Android fixes (#3988)
8a1347f1a 2020-11-08 Bumping tracy to disable vsync capture and expose the dbghelp lock.
f8b466e20 2020-10-02 Bumping tracy version and fixing API usage. (#3328)
56d44bf93 2020-05-15 Merging early work on Tracy feature branch.
```

rather than accept the other tradeoffs in my original comment (.yml complexity, runtime on large machines for every CI run, etc.)

Contributing builds upstream and then using them would certainly be nice, but I understand how that is quite a jump.

Reply:
Done. Using prebuilt tracy-capture from GCS.